### DISCUSSION TITLE
Handling Daylight Saving Time in Datetime field?

### DISCUSSION THREAD

**Discussion by tropcicstefan** [OrchardCMS/OrchardCore#10535] [Q&A]
As the title states, what is community's recommendation on handling datetime in the moment of change, custom driver with own logic?
Problem is NodaTime throws `AmbiguousTimeException ` at `LocalClock.ConvertToUtcAsync` because of `AtStrictly`.

Maybe admins should be able to define how to resolve it site wide in general settings?

**Comment by Skrypt** [comment]
I'm trying to understand. Here, when you store a Datetime for a field it is persisted as a UTC DateTime. So, it doesn't have any daylight saving time applied. It implies daylight saving time only if the timezone has one. You're not supposed to require to apply it unless your Datetime Field is used inside a calendar as a recurrent event and that you want it to apply a different offset when the DateTime is displayed in a specific TimeZone that has a daylight saving time period.

So let's say I'm in GMT-5 and I'm saving 01/01/2021 12:00 AM then it will be saved as 01/01/2021 7:00 AM
Here, we know that the TimeZone is GMT-0. So, what you need is to convert it to a specific TimeZone using NodaTime.
Then, NodaTime will know that 01/01/2021 is not in daylight saving time after applying that specific TimeZone on your UTC DateTime.

There is a code snippet here I found : https://gist.github.com/justinsaraceno/c89923ab2374b41c7069

If NodaTime throws an Exception on ConvertToLocalAsync it is because you sent a DateTime instead of a DateTimeOffset to the method.

**Comment by tropcicstefan** [comment]
Repro:
Site settings - default time zone GMT +01 Europe/Warsaw (+02:00)
Create new content type with one datetime field standard editor
Create new item with value in picker "2021-10-31 02:06" (AM)
Publish content item
DateTimeFieldDisplayDriver in UpdateAsync calls ConvertToUtcAsync where nodatime cannot handle that value

**Reply by Skrypt** [reply]
Ok, will try to repro.
